### PR TITLE
[OpenCL] Registers L2Loss

### PR DESCRIPTION
**Pull request by jeffmak**

### CODE REVIEW

File: tensorflow/core/kernels/l2loss_op.cc [hunk 1]
```diff
@@ -28,6 +28,9 @@ namespace tensorflow {
 
 typedef Eigen::ThreadPoolDevice CPUDevice;
 typedef Eigen::GpuDevice GPUDevice;
+#ifdef TENSORFLOW_USE_SYCL
+typedef Eigen::SyclDevice SYCLDevice;
+#endif // TENSORFLOW_USE_SYCL
```
Comment (lukeiwanski) [inline tensorflow/core/kernels/l2loss_op.cc:33]: tiny: there should be two spaces before //

**Review by lukeiwanski — CHANGES_REQUESTED** [review]

**Comment by keryell** [comment]
Please edit the history and `git push --force` again (good git exercise... :-) ) to fix the spelling in the commit message or we will have some hard time to get this upstreamed...

Thanks.

**Review by lukeiwanski — APPROVED** [review]

**Comment by lukeiwanski** [comment]
That pull request seems to be doing two things. 
L2Loss and ApplyMomentum.. can you split them to two PR's ?

File: tensorflow/core/kernels/training_ops.cc [hunk 2]
```diff
@@ -2158,22 +2159,6 @@ class ApplyMomentumOp : public OpKernel {
 using CPUDevice = Eigen::ThreadPoolDevice;
 using GPUDevice = Eigen::GpuDevice;
 
-#define REGISTER_KERNELS(D, T)                                         \
```
Comment (lukeiwanski) [inline tensorflow/core/kernels/training_ops.cc:2161]: Why do we remove that?

File: tensorflow/core/kernels/training_ops.cc [hunk 3]
```diff
@@ -2195,6 +2180,38 @@ REGISTER_KERNELS(GPU, Eigen::half);
 REGISTER_KERNELS(GPU, float);
 REGISTER_KERNELS(GPU, double);
 #endif
+
+#ifdef TENSORFLOW_USE_SYCL
+#define DECLARE_SYCL_SPEC(T)                                               \
```
Comment (lukeiwanski) [inline tensorflow/core/kernels/training_ops.cc:2185]: That should be in separate PR

File: tensorflow/core/kernels/training_ops.cc [hunk 4]
```diff
@@ -2195,6 +2180,38 @@ REGISTER_KERNELS(GPU, Eigen::half);
...
+  template <>                                                             \
+  void ApplyMomentum<SYCLDevice, T>::operator()(                           \
```
Comment (lukeiwanski) [inline tensorflow/core/kernels/training_ops.cc:2187]: align the comment
